src/metabase/api/dashboard.clj

```diff
@@ -1124,72 +1123,23 @@
-                                  (map (fn [database-id]
-                                         [database-id
-                                          (lib.metadata.jvm/application-database-metadata-provider database-id)]))))
-
-        filterable-columns (into {}
-                                 (map (fn [[card-id card]]
-                                        (let [dataset-query (:dataset_query card)]
-                                          [card-id
-                                           (if (seq dataset-query)
-                                             (->> dataset-query
-                                                  (lib/query (metadata-providers (:database_id card)))
-                                                  lib/filterable-columns)
-                                             [])])))
-                                 cards)]
-    (for [{:keys [target] {:keys [card]} :dashcard} mappings
-          :let  [[_ dimension] (->> (mbql.normalize/normalize-tokens target :ignore-path)
-                                    (mbql.u/check-clause :dimension))]
-          :when dimension
-          :let  [ttag      (get-template-tag dimension card)
-                 dimension (condp mbql.u/is-clause? dimension
-                             :field        dimension
-                             :expression   dimension
-                             :template-tag (:dimension ttag)
-                             (log/error "cannot handle this dimension" {:dimension dimension}))
-                 field-id  (or
-                            ;; Get the field id from the field-clause if it contains it. This is the common case
-                            ;; for mbql queries.
-                            (lib.util.match/match-one dimension [:field (id :guard integer?) _] id)
-                            ;; Attempt to get the field clause from the model metadata corresponding to the field.
-                            ;; This is the common case for native queries in which mappings from original columns
-                            ;; have been performed using model metadata.
-                            (:id (qp.util/field->field-info dimension (:result_metadata card)))
-                            ;; Look through the card's filterable columns and see if any of them match. This is common
-                            ;; when the query has an aggregation and you want to filter on something pre-aggregation.
-                            (lib.util.match/match-one dimension [:field (field-name :guard string?) _]
-                              (->> card
-                                   :id
-                                   filterable-columns
-                                   (lib/find-matching-column (lib/->pMBQL dimension))
-                                   :id)))]
-          :when field-id]
-      {:field-id field-id
-       :op       (param-type->op (:type param))
-       :options  (merge (:options ttag)
```

Comment:
Using template tag options is simply not correct... We expect default options to be set per parameter type, not based on the native query tag!